src/topology/builder.rs

```diff
@@ -262,34 +276,53 @@ pub async fn build_pieces(
                 let (mut output, control) = Fanout::new();
                 let (mut errors_output, errors_control) = Fanout::new();
 
+                let mut input_rx = input_rx
+                    .filter(move |event| ready(filter_event_type(event, input_type)))
+                    .ready_chunks(128); // 128 is an arbitrary, smallish constant
+
+                let mut timer = crate::utilization::Timer::new();
+                let mut last_report = Instant::now();
+
                 let transform = async move {
-                    while let Some(event) = input_rx.next().await {
-                        if !filter_event_type(&event, input_type) {
-                            continue;
+                    timer.start_wait();
+                    while let Some(events) = input_rx.next().await {
+                        let stopped = timer.stop_wait();
+                        if stopped.duration_since(last_report).as_secs() >= 5 {
+                            timer.report();
```

Comment:
Could save one more timer fetch by passing in `stopped` to `report` here and above. I don't think it matters for performance, since it'll only happen once every 5 seconds. More a timestamp consistency thing than anything else.